adapt-and-extend/theming/resourceregistry.rst

```diff
@@ -18,47 +18,75 @@ Introduction to Plone 5 resources
 
 Plone 5 introduces new concepts for working with JavaScript and CSS in Plone.
 
-We finally have a solution to define JavaScript modules and their dependencies - in contrast to the previous approach, where all dependencies had to be loaded in order.
-We achieve that with the tool RequireJS, which follows the Asynchronous Module Definition (AMD) approach.
-We chose AMD over other module loading implementations (like CommonJS) because AMD can be used in non-compiled form in the browser and also compiled Through-The-Web.
-This way we can also support a "development mode" like in Plone versions prior to 5, where changes in the JavaScript sources are instantly reflected (actually they are made available a few seconds after initial browser load).
+JavaScript and AMD
+------------------
+
```

Comment:
Yes, $ and require are always in the global namespace.

On Tue, Feb 16, 2016 at 10:22 PM, Cris Ewing <notifications@github.com>
wrote:

> In adapt-and-extend/theming/resourceregistry.rst
> <https://github.com/plone/documentation/pull/518#discussion_r53080616>:
>
> > @@ -18,47 +18,75 @@ Introduction to Plone 5 resources
> >
> >  Plone 5 introduces new concepts for working with JavaScript and CSS in Plone.
> >
> > -We finally have a solution to define JavaScript modules and their dependencies - in contrast to the previous approach, where all dependencies had to be loaded in order.
> > -We achieve that with the tool RequireJS, which follows the Asynchronous Module Definition (AMD) approach.
> > -We chose AMD over other module loading implementations (like CommonJS) because AMD can be used in non-compiled form in the browser and also compiled Through-The-Web.
> > -This way we can also support a "development mode" like in Plone versions prior to 5, where changes in the JavaScript sources are instantly reflected (actually they are made available a few seconds after initial browser load).
> > +JavaScript and AMD
> > +------------------
> > +
>
> @ebrehault <https://github.com/ebrehault> Perfect! I'll add a mention of
> this up here at the top, then an example in the bundle registry.
>
> As a question, the JS in that example is dependent on $ being in the
> window namespace, right? Is that something that is always true? Even under
> the new resource registry approach? We reliably have a reference to jQuery
> at $ in the javascript global namespace?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/plone/documentation/pull/518/files#r53080616>.
>